### ISSUE TITLE
Run experiments using Lab framework

### ISSUE THREAD

**Issue by abcorrea** [prost-planner/prost#83]
Right now, we use Lab only for reporting. Ideally, we would use it also to run the experiments. There are several bugs with our current experiment script (e.g., issue #82) and using the Lab code for it would probably solve some bugs.

**Comment by thomaskeller79** [comment]
We decided to resolve this issue by:

1. creating a prostlab version, analogously to the downward lab part that comes with lab, in a [separate repository](https://github.com/prost-planner/prostlab)
2. removing the corresponding files from this repo and introducing an experiments folder that contains a common setup for experiments, just like the Fast Downward repo

These are necessary steps:
- [x] prostlab must be able to start an experiment
- [ ] a "standard Prost parser" (similar to the single-search-parser of Fast Downward) is required. We decided to first merge master into this branch, as everything that is logged now should also be parsed by default.
- [ ] Creation of reports is the other part that is still missing. We need to decide what kind of reports we require and check if can simply use the "Fast Downward" reports or if we have to adjust them in prostlab.
- [ ] remove no longer needed files from the Prost repo
- [ ] introducing an experiments folder with a first sample experiment, which also contains a first common setup for experiments

**Comment by thomaskeller79** [comment]
I have ticked all the "necessary steps" boxes in the previous post and the additional tasks from our sprint sheet and I am now at a point where I believe the issue could (almost) be merged. I ran a sample experiment, the reports can be found here:

https://ai.dmi.unibas.ch/_tmp_files/tkeller/report-default.html
https://ai.dmi.unibas.ch/_tmp_files/tkeller/report-all.html

As you can see, the ipc_benchmarks.py file must now be copied for each experiment, similar to the "common_setup.py" that is used for Fast Downward. It would also be an option to move it to prostlab, but then prostlab assumes knowledge of the structure of the prost repo and I am not sure this is the right thing to do.

Apart from that, I haven't written documentation yet because I honestly am not sure where to put it. I could add some information to the wiki, or I could write a Readme.md file. What do you think is the right thing to do? I kind of would like to have the documentation inside the repo s.t. it is possible to read without visiting the github site, but I'd also like to keep it here (I'm just not sure if the wiki is the right place, it seems to be less important than it was at bitbucket). My intuition right now is to just add the info to the wiki and open up an issue where we discuss this, does this make sense?

Please let me know your opinion on the documentation and have a look at the code, both here and in the prostlab repo.

**Comment by geisserf** [comment]
Did you forget to commit? The last update I see is from 9 days ago and I do not see the ipc_benchmarks.py file.

Apart from that, I agree with having the file in the prost repository rather than in the prostlab repository.

For the documentation, I would add a README.md into the testbed folder and also add the content of the readme into the wiki. The drawback is that whenever we update the readme we have to update it on the wiki as well, but we already have this case for the README in the main directory and we could just make it a policy that whenever we want to update any documentation we update the documentation in the repository first and then update the wiki.

**Comment by thomaskeller79** [comment]
No, the last commit is really that old. I ran experiments afterwards to see if everything is working as intended, and I didn't see any further issues that need to be fixed asap.

**Comment by thomaskeller79** [comment]
I added a draft for the documentation to the [wiki](https://github.com/prost-planner/prost/wiki/Evaluation). I'm not sure if it covers all the relevant information, but it would certainly be helpful to describe prostlab in more detail (in particular where it differs from lab). However, I'd like to discuss different documentation possibilities (docs folder in the repo, github wiki, readthedocs etc) and don't want to put too much energy into this before that.

**Comment by thomaskeller79** [comment]
I just updated the initial commit with 2 more bullet points that should be done before closing this issue.

**Comment by thomaskeller79** [comment]
All open tasks have either been implemented or we decided to leave them open and we have created new issues for them.

**Comment by geisserf** [comment]
Issue has been resolved in commit ad70a89.
